pulp_rpm/tests/functional/api/test_consume_content.py

```diff
@@ -0,0 +1,81 @@
+# coding=utf-8
+"""Verify whether package manager, yum/dnf, can consume content from Pulp."""
+import unittest
+from urllib.parse import urljoin
+
+from pulp_smash import api, cli, config
+from pulp_smash.exceptions import NoKnownPackageManagerError
+from pulp_smash.pulp3.constants import DISTRIBUTION_PATH, REPO_PATH
+from pulp_smash.pulp3.utils import (
+    gen_distribution,
+    gen_repo,
+    publish,
+    sync,
+)
+
+from pulp_rpm.tests.functional.utils import (
+    gen_rpm_publisher,
+    gen_rpm_remote,
+)
+from pulp_rpm.tests.functional.constants import (
+    RPM_PUBLISHER_PATH,
+    RPM_REMOTE_PATH,
+)
+from pulp_rpm.tests.functional.utils import set_up_module as setUpModule  # noqa:F401
+from pulp_rpm.tests.functional.utils import gen_yum_config_file
+
+
+class PackageManagerConsumeTestCase(unittest.TestCase):
+    """Verify whether package manager can consume content from Pulp."""
+
+    def test_all(self):
+        """Verify whether package manager can consume content from Pulp.
+
+        This test targets the following issue:
+
+        `Pulp #3204 <https://pulp.plan.io/issues/3204>`_
+        """
+        cfg = config.get_config()
+        try:
+            cli.PackageManager._get_package_manager(cfg)  # pylint:disable=protected-access
```

Comment:
You can use `cli.PackageManager(cfg)` and the `__init__` of that class will call the private `_get_package_manager(cfg)` and raise the same exception.

Reply:
@rochacbruno , I can update it.  I would like for us to think for a long term solution.

Reply:
In terms of **O.O** the better would be

```
# Instantiate the package manager
pkg_mgr = cli.PackageManager()  # does nothing only configure stuff the lazier the better

#  Access commands and properties of `pkg_mgr`, 
# if invalid it will lazily raise the NoKnownPackageManagerException

# which package manager is this?
pkg_mgr.name == "dnf"    # .name is a @property

# install something
pkg_mgr.install('package')
```